### ISSUE TITLE
--watch mode doesn't pick up new files 

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [postcss/postcss-cli#161]
Running `postcss -u stylelint -u postcss-reporter --no-map --watch -d ./src ./src/**/*.css` in a directory structure with `src/index.css`:
* Changes to `src/index.css` are picked up fine
* Adding `src/new.css` and then modifying it doesn't get picked up

Is this by design? Can it be fixed?

**Comment by RyanZim** [comment]
That’s a bug. We’re only watching for the change event; we should watch for the add event as well. 

PR welcome!

**Comment by RyanZim** [comment]
Plugins can pass new files that need watching via the `dependency` message. postcss-import uses it, though I'm not sure what other plugins do.

**Comment by stefanfisk** [comment]
Yep, for existing files that works. The use case I was referring to is a bit different, as the plugin might have it's own globs that should be watched for add events. As far as I understand plugins have no way of triggering a new build on their own?

**Comment by RyanZim** [comment]
Ah, no.

**Comment by joeljeske** [comment]
It would seem like the original issue of the `--watch mode doesn't pick up new files` is still relevant. 

I would think that another issue/pr would be created for plugins being able to set their own dependencies and re-triggering builds, but that this issue could be closed by listing for the add/delete event in addition to change event. Is that correct? 

I would much appreciate this issue being fixed. Thanks

**Comment by RyanZim** [comment]
@joeljeske It could be fixed by listening for the `add` event. Fixing this is non-trivial, I attempted it once and never completed it due to the complexity.

I am not paid for my work here, and am not personally affected by this bug, so this will most likely remain open until someone in the community steps up and does the work needed to fix this.

**Comment by joeljeske** [comment]
Awesome, thanks for the quick response. Hopefully I will get a chance to look at this sometime soon.

**Comment by RyanZim** [comment]
@1st8 No, it still doesn't pick up new files in the given directory; it just scans the directory for files that currently exist, and watches them.

**Comment by RyanZim** [comment]
Actually, sorry, I was wrong. It watches the directory correctly. However, this original issue still isn't fixed; we don't detect new files specified in the original glob passed on the CLI.

**Comment by RyanZim** [comment]
`chokidar` (our watching library) v4 removed support for watching globs, so after we upgrade (https://github.com/postcss/postcss-cli/pull/501), this will be impossible to support. I think we just chalk this one down as a wontfix edgecase.
